Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -382,45 +375,64 @@ extension ActivityStreamPanel {
         let touchPoint = longPressGestureRecognizer.locationInView(self.view)
         guard let indexPath = tableView.indexPathForRowAtPoint(touchPoint) else { return }
 
+        var contextMenu: ActionOverlayTableViewController? = nil
+
         switch Section(indexPath.section) {
         case .Highlights:
-            contextMenuForHighlightCellWithIndexPath(indexPath)
+            contextMenu = contextMenuForHighlightCellWithIndexPath(indexPath)
         case .TopSites:
             let topSiteCell = self.tableView.cellForRowAtIndexPath(indexPath) as! ASHorizontalScrollCell
             let pointInTopSite = longPressGestureRecognizer.locationInView(topSiteCell.collectionView)
             guard let topSiteIndexPath = topSiteCell.collectionView.indexPathForItemAtPoint(pointInTopSite) else { return }
-            contextMenuForTopSiteCellWithIndexPath(topSiteIndexPath)
+            contextMenu = contextMenuForTopSiteCellWithIndexPath(topSiteIndexPath)
         case .HighlightIntro:
             break
         }
+
+        if let contextMenuToShow = contextMenu {
+            presentContextMenu(contextMenuToShow)
+        }
```

Comment:
This is most of the refactoring I've done here. Instead of combining the creation and presentation of the overlay view controller, I've separated them so I can get a handle to the view controller without worrying about it being displayed.